### PR TITLE
Add Warning About File Descriptor Usage

### PR DESCRIPTION
**Pull request by bbernays**
fixes: https://github.com/cloudquery/cloudquery/issues/285

### CODE REVIEW

**Review by yevgenypats — APPROVED** [review]

File: provider/schema/execution.go [hunk 1]
```diff
@@ -197,6 +198,9 @@ func (e ExecutionData) callTableResolve(ctx context.Context, client ClientMeta,
 		err := e.Table.Resolver(ctx, client, parent, res)
 		if err != nil && e.Table.IgnoreError != nil && e.Table.IgnoreError(err) {
 			client.Logger().Warn("ignored an error", "err", err, "table", e.Table.Name)
+			if strings.Contains(err.Error(), ": socket: too many open files") {
+				client.Logger().Warn("try increasing number of available file descriptors via `ulimit -n 10240` or by increasing timeout via provider specific parameters")
+			}
```
Comment (roneli) [inline provider/schema/execution.go:203]: The problem with this error which is okay, its not enough, the user will need to dig in the log to find the warning. Also this code will only enter if it is ignored. I suggest restructuring the if section, also setting the error with `diag.FromError` as a a Throttle diagnostic with severity warning and description etc' so user can fix this

**Review by roneli — CHANGES_REQUESTED** [review]
Good start, but we can improve this by returning a Diagnostic to the user.

File: provider/schema/execution.go (hunk 1 above)
Comment (bbernays) [reply provider/schema/execution.go]: Ok! I think I implemented what you described. Let me know if it is not what you are looking for

**Review by roneli — APPROVED** [review]
LGTM, minor nitpicks

**Pull request merged**